File: leetcode_75/Two_Pointers/cpp/392.cpp
#include <iostream>
#include <string>

using namespace std;

class Solution
{
public:
    bool isSubsequence(string s, string t)
    {
        int l1 = s.size();
        int l2 = t.size();
        if (l1 > l2)
        {
            return false;
        }
        if(l1 == 0)
        {
            return true;
        }

        int j = 0;
        for (int i = 0; i < l2; ++i)
        {
            if (t[i] == s[j])
            {
                j += 1;
            }
            if (j == l1)
            {
                return true;
            }
        }

        return false;
    }
};

File: leetcode_75/Two_Pointers/python/392.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        l1 = len(s)
        l2 = len(t)
        if l1 > l2:
            return False

        if l1 == 0:
            return True

        j = 0
        for i in range(l2):
            if t[i] == s[j]:
                j += 1
            if j == l1:
                return True

        return False


solution = Solution()
print(solution.isSubsequence(s="abc", t="ahbgdc"))
